common/lib/capa/capa/responsetypes.py

```diff
@@ -364,14 +369,22 @@ def make_hint_div(self, hint_node, correct, student_answer, question_tag,
 
         # Form the div-wrapped hint texts
         hints_wrap = u''.join(
-            [u'<div class="{0}">{1}</div>'.format(QUESTION_HINT_TEXT_STYLE, dct.get('text'))
-             for dct in hint_log]
+            [u'<div class="{question_hint_text_style}">{text}</div>'.format(
+                question_hint_text_style=QUESTION_HINT_TEXT_STYLE,
+                text=dct.get('text')
+            ) for dct in hint_log]
         )
         if multiline_mode:
-            hints_wrap = u'<div class="{0}">{1}</div>'.format(QUESTION_HINT_MULTILINE, hints_wrap)
+            hints_wrap = u'<div class="{question_hint_multiline}">{hints_wrap}</div>'.format(
+                question_hint_multiline=QUESTION_HINT_MULTILINE,
+                hints_wrap=hints_wrap
+            )
         label_wrap = ''
         if label:
-            label_wrap = u'<div class="{0}">{1}: </div>'.format(QUESTION_HINT_LABEL_STYLE, label)
+            label_wrap = u'<span class="{question_hint_label_style}">{label}: </span>'.format(
+                question_hint_label_style=QUESTION_HINT_LABEL_STYLE,
+                label=label
```

Comment:
As mentioned above, the HTML should be wrapped with HTML, and label should be wrapped with Text.